borg/remote.py

```diff
@@ -316,22 +316,18 @@ def handle_error(error, res):
             if w:
                 while not self.to_send and (calls or self.preload_ids) and len(waiting_for) < 100:
                     if calls:
-                        if is_preloaded:
-                            if calls[0] in self.cache:
-                                waiting_for.append(fetch_from_cache(calls.pop(0)))
+                        args = calls.pop(0)
+                        if cmd == 'get' and args in self.cache:
+                            waiting_for.append(fetch_from_cache(args))
                         else:
-                            args = calls.pop(0)
-                            if cmd == 'get' and args in self.cache:
-                                waiting_for.append(fetch_from_cache(args))
-                            else:
-                                self.msgid += 1
-                                waiting_for.append(self.msgid)
-                                self.to_send = msgpack.packb((1, self.msgid, cmd, args))
+                            self.msgid += 1
+                            waiting_for.append(self.msgid)
+                            self.to_send = msgpack.packb((1, self.msgid, cmd, args))
                     if not self.to_send and self.preload_ids:
                         args = (self.preload_ids.pop(0),)
                         self.msgid += 1
                         self.cache.setdefault(args, []).append(self.msgid)
-                        self.to_send = msgpack.packb((1, self.msgid, cmd, args))
+                        self.to_send = msgpack.packb((1, self.msgid, 'get', args))
```

Comment:
This if for:
"""
using anything other that a get or get_many call on the RemoteRepository would have tried to
use the per instance preload list with a wrong command.
"""

Looks correct.